admin/main.go

```diff
@@ -185,6 +185,59 @@ func verifyIDToken(app *firebase.App, idToken string) *auth.Token {
 	return token
 }
 
+// ==================================================================
+// https://firebase.google.com/docs/auth/admin/manage-sessions
+// ==================================================================
+
+func revokeRefreshTokens(app *firebase.App, uid string) {
+
+	// [START revoke_tokens]
+	ctx := context.Background()
+	client, err := app.Auth(ctx)
+	if err != nil {
+		log.Fatalf("error getting Auth client: %v\n", err)
+	}
+	if err := client.RevokeRefreshTokens(ctx, uid); err != nil {
+		log.Fatalf("error revoking tokens for user: %v, %v\n", uid, err)
+	}
+	// N.B the database API is not yet implemented in go, check the state of
+	// the tokens as not revoked with VerifyIDTokenAndCheckRevoked().
+
+	// accessing the user's TokenValidAfter
+	u, err := client.GetUser(ctx, uid)
+	if err != nil {
+		log.Fatalf("error getting user %s: %v\n", uid, err)
+	}
+	log.Printf("the refresh tokens were revoked at: %d (UTC seconds) ",
+		u.TokensValidAfterMillis/1000)
+	// [END revoke_tokens]
+}
+
+func verifyIDTokenAndCheckRevoked(app *firebase.App, idToken string) *auth.Token {
+	ctx := context.Background()
+	// [START verify_id_token_and_check_revoked]
+
+	client, err := app.Auth(ctx)
+	if err != nil {
+		log.Fatalf("error getting Auth client: %v\n", err)
+	}
+	// As opposed to the other SDKs, Go has a dedicated function to check revoaction status.
+	token, err := client.VerifyIDTokenAndCheckRevoked(ctx, idToken)
+	if err != nil {
+		if err.Error() == "ID token has been revoked" {
```

Comment:
Not directly related to this PR. But I think this will look so much better if devs could just do:

`if err == auth.IDTokenRevoked {`

Something to think about going forward.